nemo/core/classes/modelPT.py

```diff
@@ -1246,3 +1245,22 @@ def set_eff_save(use_eff_save: bool):
     def use_eff_save() -> bool:
         global _MODEL_EFF_SAVE
         return _MODEL_EFF_SAVE
+
+    def load_weights_from_checkpoint(self, model_path, strict=True):
+        if model_path.endswith(".nemo"):
+            checkpoint = self.__class__.restore_from(model_path, map_location=torch.device('cpu'))
+            self.load_state_dict(checkpoint.state_dict(), strict=strict)
+        elif model_path.endswith(".ckpt"):
+            checkpoint = torch.load(model_path, map_location=torch.device('cpu'))
+            self.load_state_dict(checkpoint["state_dict"], strict=strict)
+        else:
+            raise ValueError(
+                "'{model_path}' is not a valid file as we need either a NeMo file ('.nemo') or "
+                "a PyTorch checkpoint ('.ckpt') to load weights from."
+            )
+
+    def on_pretrain_routine_start(self):
```

Comment:
Remove this option to init model from config defined path. Too many edge cases for no benefit when its better left to be done manually using the above function by the user - in either case its a one liner.

One of the most problematic cases is that the checkpoint path is not removed from the config - so when the .nemo model gets saved, it still has this path inside it. When someone restores and tries to finetune or train from scratch, even if they dont specify this flag, the old value will still be there and weights loading will fail (since on different machine, different paths).

Another issue is that this can cause an issue with model parallel models. For now, lets revert this function entirely.

Reply:
The whole point of this PR is to enable initializing the weights by using the config file which makes the current speec_to_text.py script able to fine-tune without any need to add some extra lines. If I remove it, the whole PR becomes mostly absolute as users can always loads weights with two lines of code.

We can drop it from the config to address such cases. It does not look to me a not solvable issue. But even in this state, it does not make serious issue. No one can train from scratch or fine-tune with a nemo file as it is already trained and step is not zero anymore. It just tried to load the weights if global step is zero. The only way for such a case is load the nemo model, recreate the optimizer, data loader and all stuff and etc. We have the same issue with tokenizer, data loaders, load_from_checkpoint too.

Reply:
> No one can train from scratch or fine-tune with a nemo file as it is already trained and step is not zero anymore

.nemo files are the **primary** way to finetune or scratch train - there are multiple examples all throughout notebooks of exactly this scenario. The step is exactly 0 for finetuning - since the PTL optimizer state is not a part of .nemo files!

And yes, this should most definitely be removed from config because it will break fine tuning after restoration (reasons mentioned in the previous message)